src/ansys/fluent/core/session.py

```diff
@@ -319,6 +335,18 @@ def __dir__(self):
             )
         )
 
```

Comment:
are `_upload` and `_download` simply non-throwing no-op methods if pim is unconfigured? That is the requirement.

Reply:
Yes.  
In following usage, we do not have single dependency on PIM or it's instance, it's doesn't matter whether PIM is installed or not. 

Here ```self._remote_instance``` is act as actual ```self._pim_instance```.

```
import ansys.fluent.core as pyfluent
from ansys.fluent.core import launch_fluent

meshing_session = launch_fluent(cleanup_on_exit=False, meshing_mode=True)
pyfluent.session.Session._uploader=None
pyfluent.session.Session._upload=_upload
pyfluent.session.Session._download=_download
meshing_session._upload('upload/exhaust_system11.ipynb', remote_file_name='exh6.ipynb')
meshing_session._download('exh6.ipynb', local_file_path='download')
```